### ISSUE TITLE
Provide a blind insertion that notifies of overwrite

### ISSUE THREAD

**Issue by whitfin** [jeromefroe/lru-rs#43]
Right now if I do `cache.put("", "")`, I have no idea if the key existed already and was updated. This information is available, and it would be easy to return the previous value to the caller if the key had already existed. 

It enables this pattern:

```rust

if let Some(_) = cache.put("key", "value") {
  // the key was already in there
}
```

Whereas you would currently have to do this:

```rust
let value = cache.get("key");
if value.is_some() {
  // the key was already in there
}
cache.put("key", "value");
```

Which is pretty useful for anyone using this cache as a lookup table for things like duplication, etc.

**Comment by jeromefroe** [comment]
Thanks for the issue! This would certainly be a great ergonomic improvement and matches the [`insert` method of the standard library's `Hashmap`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert). I'll aim to tackle this issue within the next two weeks, time permitting.

**Comment by jeromefroe** [comment]
@whitfin I just made a PR which should address this issue. If you wouldn't mind taking a look when you get a chance just to confirm that the changes are in line with what you had in mind, I can then merge the PR and cut a new release.

**Comment by whitfin** [comment]
@jeromefroe that change looks good to me!

**Comment by jeromefroe** [comment]
Just [released version 0.1.17 to crates.io](https://crates.io/crates/lru/) which contains these changes.

**Comment by whitfin** [comment]
@jeromefroe perfect, thank you :D I'll try it out today
